R/hugo.R

```diff
@@ -190,19 +190,32 @@ install_theme = function(theme, theme_example = FALSE, update_config = TRUE, for
 #'   (e.g. a post or a page).
 new_content = function(path, kind = 'default', open = interactive()) {
   if (missing(kind)) kind = default_kind(path)
-  hugo_cmd(c('new', shQuote(path), c('-k', kind)))
-  file = content_file(path)
-  hugo_toYAML(file)
+  ## Hugo doesn't work well with non .md file terminations
+  ## So for those cases we will temporarely change the file termination
+  ## to md, run hugo, then change back the file termination.
+  path_temp = gsub(paste0(tools::file_ext(path), '$'), 'md', path)
+  file.copy(path, path_temp)
+  if(path_temp != path) on.exit(unlink(path_temp), add = TRUE)
+  hugo_cmd(c('new', shQuote(path_temp), c('-k', kind)))
+
+  file_temp = content_file(path_temp)
+  if(path_temp != path) on.exit(unlink(file_temp), add = TRUE)
+  ## If we do this after fixing the file termination it doesn't work
+  hugo_toYAML(file_temp)
+  file = gsub('md$', tools::file_ext(path), file_temp)
+  file.copy(file_temp, file)
```

Comment:
Basically I had written the full code in the comment (meant to replace your lines 193-206 above): https://github.com/rstudio/blogdown/pull/263#issuecomment-368724124 Did you test it? (I was asking if that would work)

```r
# create a .md file and rename it to the expected extension later
path2 = with_ext(path, '.md')
file  = content_file(path)
file2 = content_file(path2)
hugo_cmd(c('new', shQuote(path2), c('-k', kind)))
hugo_toYAML(file2)
file.rename(file2, file)
```